server/services/MonitorService.js

```diff
@@ -4,77 +4,565 @@
  */
 
 import _ from 'lodash';
+import querystring from 'querystring';
 
 import { INDEX } from '../../utils/constants';
 import { isIndexNotFoundError } from './utils/helpers';
 import { MDSEnabledClientService } from './MDSEnabledClientService';
 import { DEFAULT_HEADERS } from "./utils/constants";
 
+const isNoHandlerError = (err) =>
+  err && (
+    err.response?.includes?.('no handler found for uri') || 
+    err.body?.error?.includes?.('no handler found for uri') ||
+    err.message?.includes?.('no handler found for uri') ||
+    String(err).includes('no handler found for uri')
+  );
+
+const isLegacyMonitorDeleteError = (err) => {
+  const reason = err?.body?.error?.reason || err?.message || String(err || '');
+  return typeof reason === 'string' && reason.includes('Alerting V1 Monitor');
+};
+
+const isV2MonitorPayload = (body) =>
+  !!body?.ppl_monitor ||
+  body?.query_language === 'ppl' ||
+  body?.monitor?.query_language === 'ppl';
+
 export default class MonitorService extends MDSEnabledClientService {
+  buildAlertsPath(query = {}, { omitDataSourceId = false } = {}) {
+    const queryCopy = { ...query };
+    if (omitDataSourceId) {
+      delete queryCopy.dataSourceId;
+    }
+    const queryString = querystring.stringify(queryCopy);
+    return `/_plugins/_alerting/v2/monitors/alerts${queryString ? `?${queryString}` : ''}`;
+  }
+
+  normalizeAlertsQuery(query = {}) {
+    const {
+      from,
+      size,
+      sortField,
+      sortDirection,
+      search,
+      severityLevel,
+      monitorIds,
```

Comment:
Is this something new or existing behavior to support both array and a string as possible input?